app/models/provider.rb

```diff
@@ -310,4 +315,8 @@ def services
       )
     end
   end
+
+  def set_provider_name_search
+    self.provider_name_search = self.provider_name.downcase.gsub(/[^0-9a-z]/i, "") if self.provider_name
```

Comment:
Regex could be extracted (repeated on line 116)

Reply:
Not sure I follow you. This method is to normalise the data that we are going to store. Line 116 is normalising the query string the user inputted.

Reply:
`/[^0-9a-z]/i` could be extracted to a constant?

Reply:
I don't think we need this as a constant as its no longer in the same file.

Reply:
I think @TobyRet is right. I'd go further, there should be a single method to do normalisation, and if possible used for both generating data for this column, and normalising the query when used to search.

Reply:
+1 for extracting this. I'd make it a service which could be used here and in the controllers etc. We could unit test it then and get rid of all of the more expensive request specs that are really just testing that this works.

Reply:
Extract out into another PR #1324